Libs/MRML/Core/vtkMRMLSliceNode.cxx

```diff
@@ -339,25 +379,22 @@ void vtkMRMLSliceNode::SetOrientation(const char* orientation)
     {
     return;
     }
- if (!strcmp(orientation, "Axial"))
-    {
-    this->SetOrientationToAxial();
-    }
-  else if (!strcmp(orientation, "Sagittal"))
-    {
-    this->SetOrientationToSagittal();
-    }
-  else if (!strcmp(orientation, "Coronal"))
-    {
-    this->SetOrientationToCoronal();
-    }
-  else if (!strcmp(orientation, "Reformat"))
-    {
-    this->SetOrientationToReformat();
-    }
-  else
+
+  if (!strcmp(orientation, "Reformat"))
+   {
+   this->SetOrientationToReformat();
+   return;
+   }
+
+  this->SliceToRAS->DeepCopy(this->GetSliceOrientationPreset(orientation));
+
+  //SetOrientationString() and SetOrientationReference()
+  //behaviour will be mantained for backward compatibility
+  if(this->SliceToRAS)
     {
-    vtkErrorMacro("SetOrientation: invalid orientation: " << orientation);
+    this->SetOrientationString(orientation);
+    this->SetOrientationReference(orientation);
```

Comment:
You (or I) need to modify the rotate to volume plane. It would be great if you could give it a go and let me know if you need help or don't have time.